user/build-stages.md

```diff
@@ -0,0 +1,169 @@
+in the `jobs.include` section of your `.travis.yml` file:
+
+```yaml
+jobs:
+  include:
+    - stage: test
+      script: ./test 1
+    - stage: test
+      script: ./test 2
+    - stage: deploy
+      script: ./deploy
+```
+
+The previous yaml creates a build with three jobs, two of which start in
+parallel in the first stage (named `test`), while the third job on the second
+stage (named `deploy`) starts only after the test stage completes successfully.
+
+This screencast demonstrates how the two stages work:
+
+![Example screencast](https://cloud.githubusercontent.com/assets/3729517/25229553/0868909c-25d1-11e7-9263-b076fdef9288.gif)
+
+## Naming your stages
+
+Stage names are arbitrary strings, and can include spaces or emojis. The first
+letter of a stage name is automatically capitalized for aesthetical reasons, so
+you don't have to deal with uppercase strings in your `.travis.yml` file.
+
+The default stage is `test`. Jobs that do not have a stage name are assigned to
+the previous stage name if one exists, or the default stage name if there is no
+previous stage name. `test`. This means that if you set the stage name on the
+first job of each stage, the build will work as expected.
+
+For example the following config is equivalent to the one above, but also adds a
+second deploy job to the `deploy` stage that deploys to a different target. As
+you can see you only need to specify the stage name once:
+
+```yaml
+jobs:
+  include:
+    - script: ./test 1 # uses the default stage name "test"
+    - script: ./test 2
+    - stage: deploy
+      script: ./deploy target-1
+    - script: ./deploy target-2
+```
+
+## Build stages and build matrix expansion
+
+[Matrix expansion](https://docs.travis-ci.com/user/customizing-the-build/#Build-Matrix)
+means that certain top level configuration keys expand into a matrix of jobs.
+
+For example:
+
+```yaml
+rvm:
+  - 2.3
+  - 2.4
+jobs:
+  include:
+    - stage: deploy
+      rvm: 2.4
+      env:
+        - FOO=foo
+      script: ./deploy
+```
+
+This will run two jobs with on Ruby 2.3 and 2.4 respectively first, and assign
+these to the default stage test. The third job on the deploy stage starts only
+after the test stage has completed successfully. Be sure to set the set the
+`rvm` key on your included deploy job, too.
+
+## Build stages and deployments
+
+You can combine build stages with [deployments](https://docs.travis-ci.com/user/deployment/):
+
+```yaml
+jobs:
+  include:
+    - script: ./test 1 # uses the default stage name "test"
+    - script: ./test 2
+    - stage: deploy
+      script: skip     # usually you do not want to rerun any tests
+      deploy: &heroku
+        provider: heroku
+        # ...
+```
+
+Travis CI does not set or overwrite any of your scripts, and most languages
+have a [default test script](https://docs.travis-ci.com/user/languages/ruby/#Default-Test-Script)
+defined. So in many use cases you might want to overwrite the `script` by
```

Comment:
I like this a lot! We got a question in Support about skipping the `install` step in one of the stages. Would it make sense to mention that it's possible to skip/redefine any of the "stages" (`build`, `script`...) of the build in a Build Stage?